### PR TITLE
Implemented API end points for Downloadable File URLs

### LINKED ISSUE #66: Implement API Endpoints for Downloadable File URLs
We should let API consumers use the file download ticketing system to get downloadable file URLs.

Will need to be updated after #38.

Refs #24

### PR DESCRIPTION
**Pull request by ehlertjd**
Closes #66 

@jenreiter Could you please try this with matlab when you get a chance?

### CODE REVIEW

File: api/download.go [hunk 1]
```diff
@@ -16,6 +17,11 @@ type DownloadSource struct {
 	Path   string
 }
 
+// DownloadTicket is retrieved when a keyless download is required
```
Comment (kofalt) [inline api/download.go]: Suggest changing verbiage to mention explicitly that the download is without authentication (there are a lot of map/key concepts in our nouns & endpoints)

File: tests/download_test.go [hunk 2]
```diff
@@ -63,3 +73,24 @@ func (t *F) downloadText(fn func(string, string, *api.DownloadSource) (chan int6
...
+	buffer, dest := DownloadSourceToBuffer()
+	downloadUrl, _, err := fn(id, filename)
+
+	t.So(err, ShouldBeNil)
+	t.So(downloadUrl, ShouldNotEqual, "")
+
+	noAuthClient := api.Client{
+		t.Doer,
+		t.Sling.New().Set("Authorization", ""),
+	}
```
Comment (kofalt) [inline tests/download_test.go]: This is valid, but relies upon the ordering of struct members. It's more robust and self-documenting to reference the struct members by name when instantiating.

**Comment by ehlertjd** [comment]
@kofalt Comments addressed.

**Comment by jenreiter** [comment]
The Matlab method calls work, two things to improve:

1. Return the full URL, not the relative path
2. Update the testDrive.m file for Matlab with new methods

**Comment by kofalt** [comment]
I'll get Justin a Matlab license tomorrow, are you comfortable adding what's needed to `testDrive.m

**Comment by jenreiter** [comment]
@ehlertjd I tested the testDrive.m file - looks good. I have two comments

1. There are actually two testDrive.m files - one in 'matlab' and one in 'matlab-binary'. I don't know why we have two... We probably should revisit that decision and clean it up. 
2. Within the testDrive.m I would add semi colons at the end of the fw.getContainerDownloadUrl lines to suppress the output...

**Comment by kofalt** [comment]
The duplication stems from `matlab` vs `matlab-binary`... it's not great. I'd be fine with having it somewhere else, and then copied to both locations.

Alternatively, we could combine the two and conditionally run what's needed? That might be cleaner. Of course, the hope is to one day retire the binary approach entirely...

**Comment by ehlertjd** [comment]
@jenreiter Matlab comments addressed.

**Pull request merged**